tool/tsh/tsh.go

```diff
@@ -1926,23 +1926,32 @@ func getClusterClients(cf *CLIConf) ([]*clusterClient, error) {
 		)
 		defer span.End()
 
+		logger := log.WithField("cluster", profile.Cluster)
+
+		logger.Debug("Creating client...")
 		proxy, err := tc.ConnectToProxy(ctx)
 		if err != nil {
-			return trace.Wrap(err)
+			// log error and return nil so that results may still be retrieved
+			// for other clusters.
+			logger.Errorf("Failed connecting to proxy: %v", err)
+			return nil
 		}
 
 		sites, err := proxy.GetSites(ctx)
 		if err != nil {
-			return trace.Wrap(err)
+			// log error and create a site for the proxy we were able to
+			// connect to so that results are still retrieved.
+			logger.Errorf("Failed to lookup leaf clusters: %v", err)
+			sites = []types.Site{{Name: profile.Cluster}}
```

Comment:
What do you think about always logging these errors to stderr?

Right now they'll appear only if the user also provides the `-d` flag, otherwise the error will be silent. Assuming a cluster is down and I don't know how many resources it has (or if any), if I do `tsh ls -R` and I won't see any nodes from that cluster, I'll treat the silent failure as success and assume that the cluster doesn't have any nodes.

Reply:
Yeah, this is pretty nice!